examples/src/main/java/mil/nga/giat/geowave/examples/query/SpatialQueryExample.java

```diff
@@ -0,0 +1,497 @@
+		sfBuilder.set("geometry", GeometryUtils.GEOMETRY_FACTORY.createPoint(new Coordinate(-118.26713562011719, 33.988349152677955)));
+		sfBuilder.set("filter", "Complex-LA");
+		sfBuilder.set("latitude", 33.988349152677955);
+		sfBuilder.set("longitude", -118.26713562011719);
+		SimpleFeature basicPoint2 = sfBuilder.buildFeature("2");
+
+		ArrayList<SimpleFeature> features = new ArrayList<SimpleFeature>();
+		features.add(basicPoint1);
+		features.add(basicPoint2);
+
+		//Ingest the data. For that purpose, we need the feature adapter,
+		// the index type (the default spatial index is used here),
+		// and an iterator of SimpleFeature
+		dataStore.ingest(sfAdapter, IndexType.SPATIAL_VECTOR.createDefaultIndex(), features.iterator());
+
+		/**
+		 * After ingest, a single point might look like this in Accumulo.
+		 */
+//        \x1F\x11\xCB\xFC\xB6\xEFT\x00\xFFcomplex_feature4\x00\x00\x00\x0E\x00\x00\x00\x01\x00\x00\x00\x00 complex_feature:filter []    Complex-LA
+//        \x1F\x11\xCB\xFC\xB6\xEFT\x00\xFFcomplex_feature4\x00\x00\x00\x0E\x00\x00\x00\x01\x00\x00\x00\x00 complex_feature:geom\x00\x00 []    \x00\x00\x00\x00\x01\xC0]\x91\x18\xC0\x00\x00\x00@@\xFE\x829\x9B\xE3\xFC
+//        \x1F\x11\xCB\xFC\xB6\xEFT\x00\xFFcomplex_feature4\x00\x00\x00\x0E\x00\x00\x00\x01\x00\x00\x00\x00 complex_feature:latitude []    @@\xFE\x829\x9B\xE3\xFC
+//        \x1F\x11\xCB\xFC\xB6\xEFT\x00\xFFcomplex_feature\x00\x00\x00\x0E\x00\x00\x00\x01\x00\x00\x00\x00 complex_feature:longitude []    \xC0]\x91\x18\xC0\x00\x00\x
+	}
+
+	/**
+	 * This query will search all points using the world's Bounding Box
+	 */
+	private void pointQueryCase1() throws ParseException {
+        log.info("Running Point Query Case 1");
+		//First, we need to obtain the adapter for the SimpleFeature we want to query.
+		// We'll query basic-feature in this example.
+		//Obtain adapter for our "basic-feature" type
+		ByteArrayId bfAdId = new ByteArrayId("basic-feature");
+		FeatureDataAdapter bfAdapter = (FeatureDataAdapter) adapterStore.getAdapter(bfAdId);
+
+		//Define the geometry to query. We'll find all points that fall inside that geometry
+		String queryPolygonDefinition = "POLYGON (( " +
+				"-180 -90, " +
+				"-180 90, " +
+				"180 90, " +
+				"180 -90, " +
+				"-180 -90" +
+				"))";
+		Geometry queryPolygon = new WKTReader(JTSFactoryFinder.getGeometryFactory()).read(queryPolygonDefinition);
+
+		//Perform the query.Parameters are
+		/**
+		 * 1- Adapter previously obtained from the feature name.
+		 * 2- Default spatial index.
+		 * 3- A SpatialQuery, which takes the query geometry - aka Bounding box
+		 * 4- Filters. For this example, no filter is used.
+		 * 5- Limit. Same as standard SQL limit. 0 is no limits.
+		 * 6- Accumulo authorizations. For our mock instances, "root" works. In a real Accumulo setting, whatever authorization is associated to the user in question.
+		 */
+
+		CloseableIterator<SimpleFeature> queryResultIterator = dataStore.query(
```

Comment:
I agree - for accumulodatastore for example, the result iterators end up wrapping an accumulo scanner and you really would want to close this iterator when you are finished which in turn closes the underlying scanner